Samples/src/OracleEmulation/Operations.qs

```diff
@@ -0,0 +1,162 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+namespace Microsoft.Quantum.Samples.OracleEmulation
+{
+    open Microsoft.Quantum.Canon;
+    open Microsoft.Quantum.Primitive;
+    open Microsoft.Quantum.Extensions.Emulation;
+
+
+    //////////////////////////////////////////////////////////////////////////
+    // Defining and using simple emulated oracles ////////////////////////////
+    //////////////////////////////////////////////////////////////////////////
+
+    // Declare an `intrinsic` oracle that is implemented in C#. See the
+    // `PermutationOracle.Register` call in the driver for its implementation.
+    operation HalfAnswer(x: Qubit[], y: Qubit[]) : Unit {
```

Comment:
This as well should probably have a Q# implementation rather than only providing a definition when run with the emulation feature.

Reply:
Strictly requiring a Q# implementation with equivalent effects as the emulated version would 

1. preclude the rapid-prototyping benefits of oracle emulation and
2. introduce a non-checkable constraint (in general you'd have to be able to run the circuit to decide whether it's equivalent to a given permutation).

If we were to only require some Q# implementation (not necessarily equivalent to the emulated version), we'd likely end up with nonequivalent implementations, where the Q# version just fails, is a trivial or buggy operation because the user only cared about the emulated version. That does not appear desirable to me.

That being said, one indeed gets the greatest benefits from emulation if one can combine it with a circuit implementation for resource estimates. We will need more samples for that mode of operation. I am preparing one such sample in the context of arithmetic operations. @msoeken suggested another nice combination with his synthesis samples. This PR is meant to demonstrate just the basic ideas for defining and using an emulated oracle. Other are to be built on top.